src/formatDate.js

```diff
@@ -8,7 +8,65 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const YEAR_SHORT_FORMAT = 'YY';
+  const YEAR_LONG_FORMAT = 'YYYY';
+  const MONTH_FORMAT = 'MM';
+  const DAY_FORMAT = 'DD';
+  const CENTURY_21 = '20';
+  const CENTURY_20 = '19';
+  const MAX_YEAR = 30;
+  const DELIMITER = fromFormat.pop();
+  const SEPARATOR = toFormat.pop();
+  const dateParts = date.split(DELIMITER);
+  const shortYearIndex = fromFormat.indexOf(YEAR_SHORT_FORMAT);
+  const longYearIndex = fromFormat.indexOf(YEAR_LONG_FORMAT);
+  const formattedDateParts = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    let index;
+    let year;
+
+    switch (toFormat[i]) {
+      case YEAR_SHORT_FORMAT:
+        index = shortYearIndex < 0 ? longYearIndex : shortYearIndex;
+        year = dateParts[index];
```

Comment:
The logic for determining the index is repeated for both YEAR_SHORT_FORMAT and YEAR_LONG_FORMAT. You could refactor this into a separate function to avoid code duplication and make the code easier to maintain.